.github/workflows/continuous-integration.yml

```diff
@@ -22,6 +22,7 @@ jobs:
   build:
     strategy:
       fail-fast: false
+      max-parallel: 5
```

Comment:
@miharugelj Have limited parallelism for `matrix` strategy ... basically, pushing a commit shouldn't run more than 15 jobs simultaneously.

Also, `develop` and `release` are special branches, and are not intended to be used for PRs. The issue is that push to `develop`/`release` triggers the complete pipeline, and also pushing a commit to an open PR branch triggers the pipeline. If you have PR open on `develop`/`release` branch, each push will trigger pipeline twice.